spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherProperties.java

```diff
@@ -43,4 +71,16 @@ public void setModules(String[] modules) {
 		return modules;
 	}
 
+	@PostConstruct
+	public void deduceModuleArguments() throws Exception {
```

Comment:
Note that the index alone is a valid pivot key.
Moreover, if we want to rely on boot to help us here, I reckon we want some known prefix for the arguments.

What about 
```
MODULES=coordinates:of:log,coordinates:of:filter,coordinates:of:time  // (notice the reversed order, which may be counter intuitive btw)
ARGUMENTS_0_LOGGER_NAME=foo
ARGUMENTS_1_EXPRESSION=true
ARGUMENTS_2_PATTERN=hh:ss
ARGUMENTS_2_FIXED_DELAY=4
```

I reckon this would bind to

```
private String[] modules;

private Map<String, Map<String, String>>> arguments;
```